scripts/remotes.lua

```diff
@@ -1,13 +1,217 @@
 -- Copyright (c) 2020 Dockmeister
 -- Copyright (c) 2023 Branko Majic
+-- Copyright (c) 2024 Bryan O'Malley
 -- Provided under MIT license. See LICENSE for details.
 
 
 -- Main implementation
 -- ===================
 
 local remotes = {}
+local overlap_tolerance = 0.9
 
+--- Calculates square distance between 2 positions
+--
+-- @param a first position
+-- @param b second position--
+--
+-- @return squared distance between the positions
+--
+function dist_sq(a, b)
```

Comment:
I'll look into it. It may be marginally for performant to only be computing dist_sq rather than distance, but if factorio is using C++ to calculate these rather than lua, it may be better to just use the baked in function. 

Alternately, using the in-built function might just be 'cleaner'.

Reply:
Yeah, I just assume that the performance gains will be minimal. I might end-up giving it a quick go to see how one can do performance testing with Lua within Factorio just for fun's sake - maybe even figuring out how many times the function gets invoked.